libreoffice-core/desktop/wasm/shared.ts

```diff
@@ -145,7 +147,7 @@ export type DocumentClientBase = {
   readonly viewId: ViewId;
   on(type: CallbackType, handler: CallbackHandler): void;
   off(type: CallbackType, handler: CallbackHandler): void;
-  newView(): DocumentClient;
+  newView(): Promise<DocumentClient | null>;
```

Comment:
```suggestion
  newView(): DocumentClient | null;
```

This shouldn't be necessary because the return type should automatically be wrapped in a promise. If you're actually returning a promise to a promise, you should flatten it to a single promise.

Reply:
@chase It doesn't look like the `DocumentClientBase` methods are wrapped in a promise, only `DocumentMethods` and `DocumentWithViewMethods`.

```typescript
export type DocumentClient<
  E extends { [k: string]: (...args: any) => any } = {},
> = {
  [K in keyof DocumentMethods]: (
    ...args: Parameters<DocumentMethods[K]>
  ) => Promise<ReturnType<DocumentMethods[K]>>;
} & {
  [K in keyof DocumentWithViewMethods]: (
    ...args: Parameters<DocumentWithViewMethods[K]>
  ) => Promise<ReturnType<DocumentWithViewMethods[K]>>;
} & DocumentClientBase &
  E;
  ```

Reply:
🙈 You're right, remind me not to do PR review immediately after I wake up